salt/_utils/kubernetes_utils.py

```diff
@@ -0,0 +1,535 @@
+"""Utility methods for manipulation of Kubernetes objects in Python."""
+import datetime
+import enum
+import operator
+from pprint import pformat
+import re
+
+from salt.utils.dictdiffer import recursive_diff
+
+try:
+    import kubernetes.client as k8s_client
+    from kubernetes.client.apis import CustomObjectsApi
+except ImportError:
+    HAS_LIBS = False
+else:
+    HAS_LIBS = True
+
+__virtualname__ = 'kubernetes'
```

Comment:
I think we should namespace this as well, in case upstream decides to create a `kubernetes` utils thing at some point.

Reply:
Right, we could namespace the `renderer` as well, no? In case they get the same idea :)